### DISCUSSION TITLE
creating an alias for "python3"

### DISCUSSION THREAD

**Discussion by MED-1996** [smartcontractkit/full-blockchain-solidity-course-py#366] [Q&A] [answered]
On Mac, when checking the version of python, I get 2 different versions....

```
mitchelldaddario@Mitchells-MacBook-Air ~ % python --version
Python 2.7.16
```

```
mitchelldaddario@Mitchells-MacBook-Air ~ % python3 --version
Python 3.10.0
```

What I want is that when I call "python --version" I get 3.10.0!

---------

Same goes for when I call "pip" vs. "pip3". 

I want all my python commands to be considered from Python 3.10.0. How do I do this?

**Comment by cromewar** [comment, marked as answer]
Create symlink to latest version

`ln -s -f /usr/local/bin/python3.10 /usr/local/bin/python`

Close and open a new terminal

and try with 

`python --version`

**Reply by MED-1996** [reply]
Hi! Thanks for getting back to me so fast!

So, I'm trying to run that command in my terminal (shown below)...

`ln -s -f /usr/local/bin/python3.10 /usr/local/bin/python`

...but I'm getting told "Permission denied". My output looks like the following

`mitchelldaddario@mitchells-air web3_py_simple_storage % ln -s -f /usr/local/bin/python3.10 /usr/local/bin/python
`
`ln: /usr/local/bin/python: Permission denied
`


How do I not have permission be denied?

**Reply by cromewar** [reply]
is because the /usr/ folder is part of root directory, just use `sudo` before the rest of the command, press enter and then type your password and it should work.

**Reply by MED-1996** [reply]
I just tried that and it still didn't work :/, it's not terrible...I've just been having to use python3/pip3 to instead of python/pip when deploying my smart contracts. Below is the code of what I tried to do in the terminal

`ln: /usr/local/bin/python: Permission denied
`
`mitchelldaddario@mitchells-air web3_py_simple_storage % sudo ln -s -f /usr/local/bin/python3.10 /usr/local/bin/python
`

`Password:
`
`mitchelldaddario@mitchells-air web3_py_simple_storage % python --version                                             
`
`Python 2.7.16
`
`mitchelldaddario@mitchells-air web3_py_simple_storage % sudo ln -s -f /usr/local/bin/python3.10 /usr/local/bin/python
`

`mitchelldaddario@mitchells-air web3_py_simple_storage % sudo ln -s -f /usr/local/bin/python3.10 /usr/local/bin/python
`

`mitchelldaddario@mitchells-air web3_py_simple_storage % python3 --version
`
`Python 3.10.0
`
`mitchelldaddario@mitchells-air web3_py_simple_storage % `

**Reply by cromewar** [reply]
it seems your mac refuses to accept the symlink, however you can try again changin the paths (ln -s -f /usr/local/bin/python3.10 /usr/local/bin/python).  Type:

`which python` press enter. then
`which ptyhon3`and press enter again,
this should give you to paths, use them on the command:

`ln -s -f newpython3 newpython`

**Reply by MED-1996** [reply]
```
mitchelldaddario@mitchells-air web3_py_simple_storage % which python
/usr/bin/python
```

```
mitchelldaddario@mitchells-air web3_py_simple_storage % which python3
/Library/Frameworks/Python.framework/Versions/3.10/bin/python3

```
`mitchelldaddario@mitchells-air web3_py_simple_storage % 
`
`mitchelldaddario@mitchells-air web3_py_simple_storage % ln -s -f /Library/Frameworks/Python.framework/Versions/3.10/bin/python3 /usr/bin/python
`
`ln: /usr/bin/python: Operation not permitted
`
`mitchelldaddario@mitchells-air web3_py_simple_storage % sudo ln -s -f /Library/Frameworks/Python.framework/Versions/3.10/bin/python3 /usr/bin/python
`
`Password:
`
`ln: /usr/bin/python: Operation not permitted
`
`mitchelldaddario@mitchells-air web3_py_simple_storage % 
`

It unfortunately still didn't work.

**Reply by cromewar** [reply]
You python 3 is on a weird location (don't know why)... Never mind, you would like to use virtualenv (Python virtual environments) anyways for your projects. I'm sorry I couldn't provide you further assistance.

**Reply by MED-1996** [reply]
It's okay thank you for the help! Installing all these things is the hardest part of coding for me. I'm slowly understanding how it all works but it's really confusing.

**Reply by cromewar** [reply]
I would have suggested you to delete Python 2, this would have worked to change the reference between python and python 3, but as far as I know I could broke your Mac so better don't. You could use virtual envs as I said, this should get rid of that problem anyways.

**Reply by MED-1996** [reply]
I'll def look into that over the next couple days...but seeing as this isn't a major issue (as I can just append "3" to the end of statements) I'm not going to worry to much more about this for now!

**Reply by cromewar** [reply]
You are right is not a big deal to use "3" I still do so on my daily bases... For now I'm going to close this thread, feel free to ask any other questions.
